apps/src/lib/levelbuilder/script-editor/ScriptEditor.jsx

```diff
@@ -251,7 +253,8 @@ class ScriptEditor extends React.Component {
       description_audience: this.state.descriptionAudience,
       description_short: this.state.descriptionShort,
       resourceLinks: this.state.teacherResources.map(resource => resource.link),
-      resourceTypes: this.state.teacherResources.map(resource => resource.type)
+      resourceTypes: this.state.teacherResources.map(resource => resource.type),
+      is_migrated: this.state.isMigrated
```

Comment:
nit: because this cannot change, I would recommend just supplying `this.props.initialIsMigrated` here, and omitting isMigrated from the state. I don't think we anticipate a situation where this field can be edited from the UI, so this move seems future-proof.

Reply:
I suppose this could become `this.props.isMigrated` once it is no longer being used to set a state variable, too.

Reply:
Doesn't this PR allow for editing in the UI? So I think we need the initial version.

Reply:
Well now I'm not seeing the setting on the script edit page?

Reply:
I may be missing something here, but it seems important that this setting can NOT be edited from the script edit page. Otherwise, once this is hooked up to new script seeding and new script editor, it could mess up a live script pretty badly if a curriculum writer enabled this by accident.

I am not sure what the best option is but it seems like an additional safeguard may be a good idea. it doesn't have to be in this PR, but before we tie the new script editor (or new script seeding) to this flag, we should probably add a safeguard. I will open a ticket for this.

Reply:
At one point I had it editable but I think I changed it so it's not editable now. It did still need to be passed to the front end and back up so I'll make the original change Dave suggested.

Big +1 to additional safe guards here. I'm happy to add them to this PR or not but it would be bad if a bug caused this field to get lost.

Reply:
https://codedotorg.atlassian.net/browse/PLAT-592

Reply:
This all makes sense. I think making it not be editable makes sense. It does add in a dev hands on step but I guess thats ok. Would we want to consider allow you to decide which editor you want when you initial create the script (on the same page where you set the script slug which also can't be changed once set)?

Reply:
good point. I had mostly been thinking about the 21-22 curriculum, which will be marked automatically by the import script. A checkbox with appropriate description on the new script edit page seems nice-to-have, but not a January launch requirement. It should be easy to migrate legacy scripts, so the consequence of not doing this seems low.

Reply:
I was thinking about the new scripts that will be created for AI/ML and Content Integration when suggesting adding the settings